From 1fe2416fcf3b299e7dcd77e5df7f5ab6b0d033c0 Mon Sep 17 00:00:00 2001
From: "Ivan R. Judson" <irjudson@gmail.com>
Date: Sat, 25 Oct 2014 07:59:17 -0700
Subject: [PATCH] Added vs2013

---
 SConscript | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/SConscript b/SConscript
index e421c15..1c655a1 100644
--- a/SConscript
+++ b/SConscript
@@ -38,7 +38,7 @@ vars.Add(PathVariable('FREE_RTOS_DIR','Directory to FreeRTOS source code', os.en
 vars.Add(PathVariable('ARM_TOOLCHAIN_DIR', 'Path to the GNU ARM toolchain bin folder', os.environ.get('ARM_TOOLCHAIN_DIR'), PathVariable.PathIsDir))
 
 if default_msvc_version:
-    vars.Add(EnumVariable('MSVC_VERSION', 'MSVC compiler version - Windows', default_msvc_version, allowed_values=('8.0', '9.0', '10.0', '11.0', '11.0Exp')))
+    vars.Add(EnumVariable('MSVC_VERSION', 'MSVC compiler version - Windows', default_msvc_version, allowed_values=('8.0', '9.0', '10.0', '11.0', '11.0Exp', '12.0', '12.0Exp')))
 
 if ARGUMENTS.get('TARG', default_target) == 'win32':
     msvc_version = ARGUMENTS.get('MSVC_VERSION')
-- 
1.9.4.msysgit.2

